### PR TITLE
Compatible with hugegraph 0.8 when restore graph data

### PR DESCRIPTION
**Pull request by zhoney**
Also:
1. add backup/restore progress
2. backup/restore default type is 'all'

implemented: #33

Change-Id: I78ceea26179e7eed6a5661b971f4ac92bc7b89ac

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/base/LocalDirectory.java [hunk 1]
```diff
@@ -88,6 +96,18 @@ public InputStream inputStream(String file) {
         return zis;
     }
 
+    private InputStream textInputStream(String file) {
+        String path = Paths.get(this.directory(), file).toString();
+        InputStream is;
+        try {
+            is = new FileInputStream(path);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/base/LocalDirectory.java]: return here

File: src/main/java/com/baidu/hugegraph/base/LocalDirectory.java [hunk 2]
```diff
@@ -88,6 +96,18 @@ public InputStream inputStream(String file) {
...
+        } catch (IOException | IllegalStateException e) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/base/LocalDirectory.java]: remove IllegalStateException

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java [hunk 3]
```diff
@@ -123,6 +123,8 @@ public void backup(List<HugeType> types) {
     }
 
     protected void backupVertices() {
+        Printer.print("Vertices backup started");
+        System.out.print("Vertices has been backup: 0\b");
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: Printer. printInBackward(string)

**Review by javeme — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 4]
```diff
@@ -444,23 +414,15 @@ public int retry() {
...
 
         @Parameter(names = {"--huge-types", "-t"},
                    listConverter = HugeTypeListConverter.class,
-                   required = true,
                    description = "Type of schema/data. " +
                                  "Concat with ',' if more than one. " +
                                  "'all' means 'vertex,edge,vertex_label," +
                                  "edge_label,property_key,index_label'")
-        public List<HugeType> types;
+        public List<HugeType> types = HugeTypeListConverter.ALL_TYPES;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java:429]: update description
and improve default value of '-d'

**Review by javeme — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 5]
```diff
@@ -370,12 +340,14 @@ public boolean force() {
...
-        public String directory = "./";
+                   description = "Directory of graph schema/data, default is " +
+                                 "'./{graphname}' in local file system " +
+                                 "or '{fs.default.name}/{graphname}' in HDFS")
+        public String directory;
 
         @Parameter(names = {"--log", "-l"}, arity = 1,
                    description = "Directory of log")
-        public String logDir = "./";
+        public String logDir = "./backup-log";
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java]: ./logs

File: src/main/java/com/baidu/hugegraph/base/Printer.java [hunk 6]
```diff
@@ -57,4 +57,21 @@ public static void printMap(String name, Map<?, ?> map) {
         }
         System.out.println("}");
     }
+
+    public static void printInBackward(String message) {
+        System.out.print(message);
+        printInBackward(0);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/base/Printer.java]: 0L

File: src/main/java/com/baidu/hugegraph/base/HdfsDirectory.java [hunk 7]
```diff
@@ -156,6 +156,22 @@ public OutputStream outputStream(String file, boolean override) {
         return zos;
     }
 
+    public static HdfsDirectory constructDir(String directory, String graph,
+                                             Map<String, String> hdfsConf) {
+        String hdfsFs = hdfsConf.get(HDFS_FS_DEFAULT_NAME);
+        E.checkArgument(hdfsFs != null && !hdfsFs.isEmpty(),
+                        "'fs.default.name' can not be null or empty " +
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/base/HdfsDirectory.java]: use HDFS_FS_DEFAULT_NAME instead

**Review by javeme — CHANGES_REQUESTED** [review]

**Review by javeme — DISMISSED** [review]

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java [hunk 8]
```diff
@@ -230,7 +230,13 @@ private void backupVertexShard(Shard shard) {
             if (vertexList == null || vertexList.isEmpty()) {
                 return;
             }
-            this.backup(HugeType.VERTEX, suffix.get(), vertexList);
+            int length = vertexList.size();
+            for (int startIndex = 0; startIndex < length; startIndex += BATCH) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: startIndex => start
endIndex => end

File: src/main/java/com/baidu/hugegraph/manager/BackupManager.java [hunk 9]
```diff
@@ -256,20 +262,26 @@ private void backupEdgeShard(Shard shard) {
             if (edgeList == null || edgeList.isEmpty()) {
                 return;
             }
-            this.backup(HugeType.EDGE, suffix.get(), edgeList);
+            int length = edgeList.size();
+            for (int startIndex = 0; startIndex < length; startIndex += BATCH) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/BackupManager.java]: why not move this loop to backup()

**Review by javeme — CHANGES_REQUESTED** [review]

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**